backend/src/main/java/com/example/subjecthub/controller/SubjectServiceController.java

```diff
@@ -69,4 +77,88 @@ public Subject getSubject(
         // TODO: Don't allow cross university fetching
         return subjectRepository.findOne(subjectId);
     }
+
+    @Override
+    public List<SubjectComment> getCommentsByUser(
+        @PathVariable Long universityId,
+        @PathVariable Long userId
+    ){
+        return subjectCommentRepository.findByUser_Id(userId);
+    }
+
+    //TODO: ALL COMMENT IMPLEMENTATIONS: as above, reflect null on no subject, no cross-uni fetch
+    @Override
+    public List<SubjectComment> getComments(
+        @PathVariable Long universityId,
+        @PathVariable Long subjectId
+    ){
+        return subjectCommentRepository.findBySubject_Id(subjectId);
+    }
+
+    @Override
+    public SubjectComment getComment(
+        @PathVariable Long universityId,
+        @PathVariable Long subjectId,
+        @PathVariable Long commentId
+    ){
+        return subjectCommentRepository.findBySubject_IdAndId(subjectId,commentId);
+    }
+
+    @Override
+    public SubjectComment commentAdd(
+        @PathVariable Long universityId,
+        @PathVariable Long subjectId,
+        @RequestBody AddCommentRequest addCommentRequest
+    ){
+        SubjectComment newComment = new SubjectComment();
+        newComment.setPost(addCommentRequest.getComment());
+        newComment.setUser(subjectHubUserRepository.findOne(addCommentRequest.getUserId()));
+        newComment.setSubject(subjectRepository.findOne(subjectId));
+        newComment.setPostTimeNow();
+        return subjectCommentRepository.save(newComment);
+    }
+
+    @Override
+    public SubjectComment commentThumbUp(
+        @PathVariable Long universityId,
+        @PathVariable Long subjectId,
+        @PathVariable Long commentId
+    ){
+        SubjectComment comment = subjectCommentRepository.findBySubject_IdAndId(subjectId,commentId);
```

Comment:
Missing null check.

Reply:
not entirely sure what you mean (I dont really know what a null check is - guessing it's a way to make sure a nullpointerexception doesnt break the system), but, I was assuming someone would have to press a 'thumb up' button from the comment itself so it's theoretically impossible for it to be null...

Reply:
Regardless of the actions made by the user from the frontend, the API itself is publicly available. Try using POSTMAN and send a request to something like /api/universities/university/1/subjects/subject/1/comments/comment/999999 and see that the app will fail because the database doesn't find that comment in the database and attempts to call a method on a null object.